cmd/thanos/sidecar.go

```diff
@@ -148,6 +163,10 @@ func runSidecar(
 						"err", err,
 					)
 					promUp.Set(0)
+					statusProber.SetNotReady(err)
+					if sinceLastSuccessfulHeartbeat >= sinceLastSuccessfulHeartbeatLimit {
+						statusProber.SetNotHealthy(err)
```

Comment:
For initial retry we either need higher limit or just let's not do it maybe (?)

Essentially you can configure easily sidecar without any hierarchy in Kuberentes - this means that this can retry for some time, as Prometheus container is starting up maybe.. I think 5m or more even if you have slowness on startup.. 

I just want to be careful to not block anyone with this (: